File: proxy/src/main/java/one/hyro/proxy/commands/KickCommand.java
package one.hyro.proxy.commands;

import com.mojang.brigadier.Command;
import com.mojang.brigadier.arguments.StringArgumentType;
import com.mojang.brigadier.tree.LiteralCommandNode;
import com.velocitypowered.api.command.BrigadierCommand;
import com.velocitypowered.api.command.CommandSource;
import com.velocitypowered.api.command.VelocityBrigadierMessage;
import com.velocitypowered.api.proxy.ProxyServer;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;

public class KickCommand {
    public static BrigadierCommand createKickCommand(final ProxyServer proxy) {
        LiteralCommandNode<CommandSource> node = BrigadierCommand.literalArgumentBuilder("kick")
                .requires(source -> source.hasPermission("hyro.kick"))
                .executes(context -> {
                    CommandSource source = context.getSource();
                    source.sendMessage(Component.translatable("info.errors.providePLayer", NamedTextColor.RED));
                    return Command.SINGLE_SUCCESS;
                })
                .then(BrigadierCommand.requiredArgumentBuilder("argument", StringArgumentType.word())
                        .suggests((ctx, builder) -> {
                            proxy.getAllPlayers().forEach(player -> builder.suggest(
                                    player.getUsername(),
                                    VelocityBrigadierMessage.tooltip(Component.text(player.getUsername()))
                            ));
                            return builder.buildFuture();
                        })
                        .executes(context -> {
                            CommandSource source = context.getSource();
                            String argumentProvided = context.getArgument("argument", String.class);

                            proxy.getPlayer(argumentProvided).ifPresentOrElse(player ->
                                            player.disconnect(Component.translatable("info.moderation.kick", NamedTextColor.RED)),
                                    () -> source.sendMessage(Component.translatable("info.errors.playerNotFound", NamedTextColor.RED))
                            );

                            Component successMessage = Component.translatable(
                                    "commands.kick.success",
                                    Component.text(argumentProvided, NamedTextColor.DARK_GREEN)
                            ).color(NamedTextColor.GREEN);

                            source.sendMessage(successMessage);
                            return Command.SINGLE_SUCCESS;
                        })
                )
                .build();

        return new BrigadierCommand(node);
    }
}

File: proxy/src/main/kotlin/commands/KickCommand.kt
package one.hyro.commands

import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.tree.LiteralCommandNode
import com.velocitypowered.api.command.BrigadierCommand
import com.velocitypowered.api.command.CommandSource
import com.velocitypowered.api.proxy.ProxyServer
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor

object KickCommand {
    fun createKickCommand(proxy: ProxyServer): BrigadierCommand {
        val node: LiteralCommandNode<CommandSource> = BrigadierCommand.literalArgumentBuilder("kick")
            .requires { source -> source.hasPermission("hyro.kick") }
            .executes { context ->
                val source: CommandSource = context.source
                source.sendMessage(Component.translatable("context.error.specifyPlayer", NamedTextColor.RED))
                return@executes 0
            }
            .then(BrigadierCommand.requiredArgumentBuilder("player", StringArgumentType.word())
                .suggests { _, builder ->
                    val players = proxy.allPlayers.map { it.username }
                    players.forEach { builder.suggest(it) }
                    builder.buildFuture()
                }
                .executes { context ->
                    val source: CommandSource = context.source
                    val player = StringArgumentType.getString(context, "player")
                    val target = proxy.getPlayer(player)

                    if (target.isPresent) {
                        target.get().disconnect(Component.translatable("context.success.kickMessage", NamedTextColor.RED))
                        val successMessage = Component.translatable(
                            "context.success.kick",
                            Component.text(player, NamedTextColor.DARK_GREEN)
                        ).color(NamedTextColor.GREEN)
                        source.sendMessage(successMessage)
                    } else {
                        source.sendMessage(Component.translatable("context.error.playerOffline", NamedTextColor.RED))
                    }
                    return@executes 1
                }
            )
            .build()

        return BrigadierCommand(node)
    }
}